File: src/main/java/com/atsuishio/superbwarfare/entity/vehicle/SpeedboatEntity.java
package com.atsuishio.superbwarfare.entity.vehicle;

import com.atsuishio.superbwarfare.entity.vehicle.base.GeoVehicleEntity;
import com.atsuishio.superbwarfare.entity.vehicle.base.WeaponVehicleEntity;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.level.Level;
import software.bernie.geckolib.animation.*;

public class SpeedboatEntity extends GeoVehicleEntity implements WeaponVehicleEntity {

    public SpeedboatEntity(EntityType<SpeedboatEntity> type, Level world) {
        super(type, world);
    }

    private PlayState machineGunFirePredicate(AnimationState<SpeedboatEntity> event) {
        if (getShootAnimationTimer(0, 0) > 0) {
            return event.setAndContinue(RawAnimation.begin().thenPlay("animation.speedboat.fire"));
        }
        return event.setAndContinue(RawAnimation.begin().thenLoop("animation.speedboat.idle"));
    }

    @Override
    public void registerControllers(AnimatableManager.ControllerRegistrar data) {
        data.add(new AnimationController<>(this, "machineGun", 0, this::machineGunFirePredicate));
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/entity/vehicle/SpeedboatEntity.kt
package com.atsuishio.superbwarfare.entity.vehicle

import com.atsuishio.superbwarfare.entity.vehicle.base.VehicleEntity
import net.minecraft.world.entity.EntityType
import net.minecraft.world.level.Level

class SpeedboatEntity(type: EntityType<SpeedboatEntity>, world: Level) : VehicleEntity(type, world), BasicGeoVehicleEntity {

//    override fun registerControllers(data: AnimatableManager.ControllerRegistrar) = buildControllers(data) {
//        "machineGun" {
//            if (getShootAnimationTimer(0, 0) > 0) {
//                thenPlay("animation.speedboat.fire")
//            } else {
//                thenLoop("animation.speedboat.idle")
//            }
//        }
//    }
}
